QUALITY_DECLARATION.md

```diff
@@ -23,17 +23,17 @@ On Git, the tags have a `v` prefix before the version numbers. A [release script
 
 `iceoryx` is at a stable version, i.e. `>= 1.0.0`. The latest valid release can be found on the [release page](https://github.com/eclipse-iceoryx/iceoryx/releases) of iceoryx.
 
-The change history can be found in the [release notes section](https://iceoryx.io/latest/release-notes).
+The change history can be found in the [release notes section](doc/website/release-notes/iceoryx-v2-0-0.md).
```

Comment:
The existing link was broken but ideally we want to link latest here. Can we do this and if so how? @dkroenke

Reply:
I will rework this accordingly, please leave the link here as-is

Reply:
@dkroenke you mean I should undo the change I presume?